### ISSUE TITLE
Support nested paths in field config specifically list type members

### ISSUE THREAD

**Issue by mbaijal** [aws-controllers-k8s/community#932]
**Is your feature request related to a problem?**
The ApplicationAutoscaling CRDs do not include the creationTime in Status even though it is a part of the describe API responses because of the list type response. An attempt to fix this by explicitly specifying the field in the generator config also fails since the `getMemberByPath` method does not support specifying nested paths as required by this field. 

Basically, I am trying to add the CreationTime to the status field as follows and this does not work as expected - 
```
ScalableTarget:
    fields:
      CreationTime:
        from: DescribeScalableTarget
        path: ScalableTargets..CreationTime
```

**Describe the solution you'd like**
Enable `getMemberByPath` to handle such list type paths.

**Comment by RedbackThomson** [comment]
Implemented as part of https://github.com/aws-controllers-k8s/code-generator/pull/183 with a follow up for `map` type in #936.
